corehq/apps/users/middleware.py

```diff
@@ -56,7 +56,7 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             # us to check location-based permissions before we can check authentication.
             # See LocationAccessMiddleware.process_view()
             username, _ = get_username_and_password_from_request(request)
-            request.couch_user = CouchUser.get_by_username(username)
+            request.couch_user = CouchUser.get_by_username(username) or InvalidUser()
```

Comment:
In line with my last comment - if `get_username_and_password_from_request` doesn't work for normal Django session auth, then this would get set as `InvalidUser()`.